pkg/environment/environment.go

```diff
@@ -30,6 +30,14 @@ func GetDefaultPaastaKubernetesEnvironment() []corev1.EnvVar {
 				},
 			},
 		},
+		{
+			Name: "PAASTA_INSTANCE_TYPE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.labels['paasta.yelp.com/service']",
```

Comment:
(copying my slack comment here): i don't think this will always be correct - for flink, there's many services that host flink jobs (i.e., there's not a single flink service with all the flink instances)